integration/cluster.go

```diff
@@ -769,11 +779,24 @@ func NewClusterV3(t *testing.T, cfg *ClusterConfig) *ClusterV3 {
 	return clus
 }
 
+func (c *ClusterV3) SetClosedClient(idx int) {
+	c.mu.Lock()
+	c.closedClientIdx[idx] = struct{}{}
+	c.mu.Unlock()
+	return
+}
+
 func (c *ClusterV3) Terminate(t *testing.T) {
-	for _, client := range c.clients {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	for i, client := range c.clients {
+		if _, ok := c.closedClientIdx[i]; ok {
```

Comment:
instead of a map, can this set the closed client to nil/check if the `client` is nil before closing?